Assets/MixedRealityToolkit.Tools/MSBuild/Scripts/AssetScriptReferenceRetargeter.cs

```diff
@@ -403,7 +403,23 @@ private static void PostProcess(string outputPath, Dictionary<string, AssemblyIn
             DirectoryInfo outputDirectory = new DirectoryInfo(outputPath);
             RecursiveFolderCleanup(outputDirectory);
             CopyPluginContents(Application.dataPath.Replace("Assets", "NuGet/Plugins"));
+
+            // Special case the Microsoft.MixedReality.Toolkit.Providers.WindowsMixedReality.dll for UNITY_WSA Editor
+            string dllPath = Utilities.GetFullPathFromAssetsRelative($"Assets/../MSBuild/Publish/InEditor/WSA/Microsoft.MixedReality.Toolkit.Providers.WindowsMixedReality.dll");
+            string pdbPath = Path.ChangeExtension(dllPath, ".pdb");
+            string editorOutputDirectory = Application.dataPath.Replace("Assets", "NuGet/Plugins/EditorPlayer");
+
+            string dllOutputPath = Path.Combine(editorOutputDirectory, "Microsoft.MixedReality.Toolkit.Providers.WindowsMixedReality.dll");
+            File.Copy(dllPath, dllOutputPath, true);
+            File.Copy(pdbPath, Path.Combine(editorOutputDirectory, "Microsoft.MixedReality.Toolkit.Providers.WindowsMixedReality.pdb"), true);
+
+            // Update metas after copying in the special cased library
             UpdateMetaFiles(assemblyInformation);
+
+            // Patch the special cased library to have a define_constraint:
+            string dllMetaPath = $"{dllOutputPath}.meta";
+            Debug.Log($"Patching: {dllMetaPath}");
+            File.WriteAllText(dllMetaPath, File.ReadAllText(dllMetaPath).Replace("defineConstraints: []", "defineConstraints:\r\n    UNITY_WSA"));
```

Comment:
@keveleigh is there any guarantee that the existing defineConstraints will always be [] going forward?

Otherwise it could be possible for this to start silently failing to do anything - (i.e. if someone defines another arbitrary constraint this will no longer do the right thing of adding in UNITY_WSA).

The ask here isn't to future proof every single possible defineConstraints content permutation (i.e. the ask isn't to parse the constraints and ensure it gets added to the list) - it's more that we should log an error (or better, just throw an exception because no one will read the error logs on CI and see this) if this assumption is violated.

Reply:
Will add a check and throw an error if the string is not found.